robot-agent/config.js

```diff
@@ -1,18 +1,72 @@
 const fs = require('fs');
 const dotenv = require('dotenv');
+const { clone } = require('@transitive-sdk/utils');
+const { getInstalledPackages, updatePackageConfigFile } = require('./utils');
 
 dotenv.config({path: './.env'});
 dotenv.config({path: './.env_user'});
 
 global.config = {};
 const fleetConfig = {};
 
-try {
-  global.config = JSON.parse(fs.readFileSync('./config.json', {encoding: 'utf8'}));
-  console.log(`Using config:\n${JSON.stringify(global.config, true, 2)}`);
-} catch (e) {
-  console.log('No config.json file found or not valid JSON, proceeding without.');
-}
+const refreshGlobalConfigFromFile = () => {
+  try {
+    global.config = JSON.parse(fs.readFileSync('./config.json', {encoding: 'utf8'}));
+    console.log(`Using config:\n${JSON.stringify(global.config, true, 2)}`);
+    const packages = getInstalledPackages();
+    // generate config.json for each package
+    packages.forEach(updatePackageConfigFile);
+    // update the config in the fleet data store
+    if (global.data) {
+      // update the config in the fleet data store
+      console.log(`Updating global config in ${global.AGENT_PREFIX}/info/config`);
+      global.data.update(`${global.AGENT_PREFIX}/info/config`, clone(global.config));
+    }
+  } catch (e) {
+    console.log('No config.json file found or not valid JSON, proceeding without.');
+  }
+};
+refreshGlobalConfigFromFile();
+
+let watcher;
+
+// Wait for the file to exist before using it
+// This is useful for VI or other editors that creates a new file
+// on save, which can cause the watcher to trigger before the file is ready
+// to be read.
+const waitForFile = (filePath, callback) => {
```

Comment:
Good point about needing to wait for the file to exists. But we should poll the filesystem twice a second for potentially every if the user never creates the file. Let's just create it as an empty JSON object, `{}`, if it doesn't exist yet.

Reply:
Alternatively we could watch the directory itself.